client/layout/guided-tours/config-elements/quit.js

```diff
@@ -27,13 +27,9 @@ export default class Quit extends Component {
 		this.removeTargetListener();
 	}
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillUpdate() {
-		this.removeTargetListener();
-	}
-
 	componentDidUpdate() {
 		this.addTargetListener();
+		this.removeTargetListener();
```

Comment:
I'm not sure I understand what's happening here. Are we adding an event listener just to remove it in the next line?

Reply:
Oh, I think you're right. I think this meant removing the old listener and attaching the new one. I re-ordered them in af7cf1bb3f3f1fcba2a39f1b53e0c561a169b7fc